src/RequestContext.js

```diff
@@ -40,8 +40,15 @@ module.exports = class RequestContext {
     this._wskActivationId = utils.randomChars(32, true);
     this._requestId = utils.randomChars(32);
     this._cdnRequestId = utils.uuid();
-    this._strain = cfg.selectStrain({ ...req, path: this._path });
-    if (this._strain.urls.length > 0) {
+
+    const reqP = { ...req, path: this._path };
+    this._strain = cfg.selectStrain(reqP);
+    if (this._strain.condition) {
+      // ugly: we have to re-execute the match op because selectStrain
+      //       just remembers the strain, not the match result
+      const match = this._strain.condition.match(reqP);
+      this._mount = match.baseURL || '';
```

Comment:
I have a question for you guys: `selectStrain()` finds the matching strain, and returns it. With conditions, the `match()` operation not only returns true/false but also the `baseURL` that matched, which is exactly what the subsequent `mount` calculation requires. Is there a way to "enhance" selectStrain to return the match result as well?

Reply:
maybe just return an info object:

```
const { match, strain } = cfg.matchStrain();
this._strain = strain;
this._mount = match.baseURL;
```

Reply:
Yes, I thought about that, but wasn't sure whether something would break, because HelixProject is an exported JS class...

Reply:
that's why I suggested a new name: `matchStrain`

Reply:
Ah, I see :)

Reply:
Thanks, @tripodsan , I now added `matchStrain` to HelixProject and all tests pass.